src/main/java/frc/robot/systems/BallHandlingFSM.java

```diff
@@ -98,21 +135,78 @@ public void update(TeleopInput input) {
 	 * @return FSM state for the next iteration
 	 */
 	private FSMState nextState(TeleopInput input) {
+		if (input == null) {
+			return FSMState.IDLE;
+		}
+		System.out.println("Velocity: " + intakeMotor.getMotorTemperature());
+
 		switch (currentState) {
 			case IDLE:
-				if (input != null && input.isShooterButtonPressed()) {
+				if (pushCommandTime == -1 && input.isShooterButtonPressed()) {
```

Comment:
please replace `-1` with a descriptive variable as it's not an obvious magic num

Reply:
replaced with much better named variables